cpu/sam0_common/periph/rtc_rtt.c

```diff
@@ -463,6 +473,16 @@ int rtc_set_alarm(struct tm *time, rtc_alarm_cb_t cb, void *arg)
     /* prevent old alarm from ringing */
     rtc_clear_alarm();
 
+#ifdef CPU_COMMON_SAMD21
+    /* When an alarm match occurs, the Alarm 0 Interrupt flag in the Interrupt
+     * Flag Status and Clear registers (INTFLAG.ALARMn0) is set on the next
+     * 0-to-1 transition of CLK_RTC_CNT. E.g. For a 1Hz clock counter, it means
+     * the Alarm 0 Interrupt flag is set with a delay of 1s after the occurrence
+     * of alarm match.
+     */
+    time->tm_sec--;
```

Comment:
Now that you mention it, I think `rtc_tm_normalize()` does not deal well with negative numbers.